### PR TITLE
7739 classloader

### PR DESCRIPTION
**Pull request by nmohamad**

### CODE REVIEW

**Comment by nmohamad** [comment]
branch is ready for promotion

File: doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+1. Abstract
+-----------
+BridgePoint Verifier uses a custom Class Loader to load classes into the execution
+session. The existing Class Loader accesses internal Eclipse components that are
+no longer available in Eclipse 4.4
+
+2. Document References
+----------------------
+[1] [BridgePoint DEI #7739](https://support.onefact.net/redmine/issues/7739)
```
Comment (keithbrown) [inline doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md]: Add 2 spaces at the end of each of these lines so you get line breaks in the rendered output

File: doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+6. Implementation Comments
+--------------------------
+See [3]
+
+7. Unit Test
+------------
+7.1 A new test model is created RealizedComponentTest under Models repo. The 
+test class contains 3 components, two of them are realized. The models cover 
+all interface communication scenarios as follows.
```
Comment (keithbrown) [inline doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md]: "...scenarios as follows" implies there will be a list of scenarios given, but there is no list.

File: doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+A java junit class RealizedComponentTest.java is created, to load the test model
+in verifier, execute 7 tests functions and compare the console log output.
+
+8. Code Changes
+---------------
+Branch name: <7739_classloader>
```
Comment (keithbrown) [inline doc-bridgepoint/notes/7739_BridgePoint_ClassLoader/7739_BridgePoint_ClassLoader.int.md]: Code changes only shows changes on the bridgepoint repository, but earlier you said there are changes to the models repo.  Those should be listed here as well along with the branch name there if it is different.

File: src/org.xtuml.bp.core/src/org/xtuml/bp/core/Vm_c.java [hunk 4]
```diff
@@ -21,9 +21,7 @@
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.core.variables.VariablesPlugin;
-/* TODO - SKB removed
-import org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader;
-*/
+//import org.eclipse.osgi.internal.baseadaptor.DefaultClassLoader;
```
Comment (keithbrown) [inline src/org.xtuml.bp.core/src/org/xtuml/bp/core/Vm_c.java]: You should remove this commented out import.

File: src/org.xtuml.bp.core/src/org/xtuml/bp/core/util/BPClassLoader.java [hunk 5]
```diff
@@ -22,35 +22,127 @@
...
+		synchronized(ClassPathEntries) {
+			if (ClassPathEntries.indexOf(path) == -1) {
+				ClassPathEntries.add(path);
+			}
+		}
 	}
 
 	public Class<?> loadClass(String name) throws ClassNotFoundException {
-	  return null;
+      Class<?> clazz = null;
```
Comment (keithbrown) [inline src/org.xtuml.bp.core/src/org/xtuml/bp/core/util/BPClassLoader.java]: The indentation in this section is really messed up.  I think you need to set the preference to "Insert spaces for tabs" and then reformat.

File: src/org.xtuml.bp.debug.ui.test/src/org/xtuml/bp/debug/test/RealizedComponentTest.java [hunk 6]
```diff
@@ -0,0 +1,440 @@
+//========================================================================
```
Comment (keithbrown) [inline src/org.xtuml.bp.debug.ui.test/src/org/xtuml/bp/debug/test/RealizedComponentTest.java]: Ick.  Delete lines 1-8.

**Pull request merged**